AsyncDisplayKit/Details/ASFlowLayoutController.mm

```diff
@@ -57,17 +57,9 @@ - (NSSet *)indexPathsForScrolling:(ASScrollDirection)scrollDirection rangeMode:(
   ASRangeTuningParameters      tuningParameters  = [self tuningParametersForRangeMode:rangeMode rangeType:rangeType];
 
   if (_layoutDirection == ASFlowLayoutDirectionHorizontal) {
-    ASDisplayNodeAssert(scrollDirection == ASScrollDirectionNone ||
-                        scrollDirection == ASScrollDirectionLeft ||
-                        scrollDirection == ASScrollDirectionRight, @"Invalid scroll direction");
-
```

Comment:
Needed to remove these because the range controller asks the layout controller about `[.down, .right]` before the user's first scroll.

It's a little bit hacky, I know, but two excuses:
- The alternative would be for the range controller to have a new code path to determine the scrollable directions. We could do it, but it's added code.
- These assertions have to go anyway. If I use a vertical flow layout, and I set `alwaysBounceHorizontally` and scroll sideways/diagonally that's a valid usage of flow layout and it would fail this assertion.